tingbot/cache.py

```diff
@@ -0,0 +1,135 @@
+#image cache
+import time
+import re
+import rfc822
+import calendar
+import requests
+import io
+import os
+from urlparse import urlparse
+
+def get_http_timestamp(dt):
+    return calendar.timegm(rfc822.parsedate(dt))
```

Comment:
I have now addressed this issue. All time comparisons should now be done in the relevant (local or server) timebase.